src/crypto/tls/cfkem_test.go

```diff
@@ -78,44 +64,43 @@ func testHybridKEX(t *testing.T, scheme kem.Scheme, clientPQ, serverPQ,
 	var expectedRetry bool
 
 	if clientPQ && serverPQ && !clientTLS12 && !serverTLS12 {
-		expectedKEX = kemSchemeKeyToCurveID(scheme)
+		expectedKEX = curveID
 	} else {
 		expectedKEX = X25519
 	}
 	if !clientTLS12 && clientPQ && !serverPQ {
 		expectedRetry = true
 	}
 
+	if expectedRetry != retry {
+		t.Errorf("Expected retry=%v, got retry=%v", expectedRetry, retry)
+	}
 	if clientSelectedKEX == nil {
 		t.Error("No KEX happened?")
-	}
-
-	if *clientSelectedKEX != expectedKEX {
+	} else if *clientSelectedKEX != expectedKEX {
```

Comment:
Reviewer note: this code was moved here to avoid a nil dereference panic when the handshake fails.